🚀 Job Board: Add the sidebar search form

DeanBiton opened the pull request:
- Created a basic sidebar
- Use a FormView for the job_board's URL that includes the search form.
- Screenshot of the webpage:
![1 jpg](https://user-images.githubusercontent.com/83355266/163876116-7a22b2b9-bd22-4349-bb4d-1b3f97e7782f.png)

Close #119 

Future tasks:
- #120
- #121
- #126
- #127

paOmer reviewed (changes requested):
I think it will be better if you'll add the job_board css link only to the job_board html templates and not to the base template or you'll completely remove the job_board css file and add the style changes you wish to have to the existing main css file.

Yarboa reviewed (commented):
@DeanBiton Why are you testing only form/views and client tests?
There is a difference here in the coverage, 
Is there any test in code that cover  the PR? in case yes please  add reference to main branch test

DeanBiton commented:
> @DeanBiton Why are you testing only form/views and client tests? There is a difference here in the coverage, Is there any test in code that cover the PR? in case yes please add reference to main branch test

@Yarboa I added a new form and a view that uses this form. I have tests for the form and added new tests for the view.

Yarboa commented:
> > @DeanBiton Why are you testing only form/views and client tests? There is a difference here in the coverage, Is there any test in code that cover the PR? in case yes please add reference to main branch test
> 
> @Yarboa I added a new form and a view that uses this form. I have tests for the form and I added more tests for the view.

There are no client tests at all for the relevant template and views

Yarboa commented on job_board/tests/test_search_view.py:
```diff
@@ -0,0 +1,38 @@
+        response = client.get(JOB_BOARD_URL)
+        assert response.status_code == 200
+
+    def test_search_using_valid_form(self, client):
```
You can add fixture for the form content, no need of static variables
Both tests returns 200OK how do you know pages generated created properly? 

What happens when non/successful form submission created? the difference should be tested

Post method in view returns this 
return render(request, 'job_board/job_board.html', context)

I would recommend to check the context

DeanBiton replied:
@Yarboa Added the relevant fixture.
Also, I am checking a successful form submission by searching for the use of the job_board/post.html template which represents a search result.

Yarboa reviewed (approved):
LGTM
